io/zenoh-link-commons/src/unicast.rs

```diff
@@ -118,3 +120,71 @@ pub fn get_ip_interface_names(addr: &SocketAddr) -> Vec<String> {
         }
     }
 }
+
+#[derive(Clone, Debug, Serialize, Hash, PartialEq, Eq)]
+pub enum LinkAuthType {
+    Tls,
+    Quic,
+    None,
+}
+
+#[derive(Clone, Debug, Serialize, Hash, PartialEq, Eq)]
+pub struct LinkAuthId {
+    auth_type: LinkAuthType,
+    auth_value: Option<String>,
+}
+
+impl LinkAuthId {
+    pub fn get_type(&self) -> &LinkAuthType {
+        &self.auth_type
+    }
+    pub fn get_value(&self) -> &Option<String> {
+        &self.auth_value
+    }
+}
+
+impl Default for LinkAuthId {
+    fn default() -> Self {
+        LinkAuthId {
+            auth_type: LinkAuthType::None,
+            auth_value: None,
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct LinkAuthIdBuilder {
+    pub auth_type: LinkAuthType,    // HAS to be provided when building
+    pub auth_value: Option<String>, // actual value added to the above type; is None for None type
+}
+
+impl Default for LinkAuthIdBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl LinkAuthIdBuilder {
+    pub fn new() -> LinkAuthIdBuilder {
+        LinkAuthIdBuilder {
+            auth_type: LinkAuthType::None,
+            auth_value: None,
+        }
+    }
+
+    pub fn auth_type(&mut self, auth_type: LinkAuthType) -> &mut Self {
```

Comment:
While different options exist for builders, e.g. to accept `&mut self` *vs* `self`. In Zenoh we tend to use the `self` version (see [here](https://github.com/eclipse-zenoh/zenoh/blob/7adad9482f8d27fab56ca54aa053de4cc2f63557/zenoh/src/subscriber.rs#L458)).